src/test/scala/mesosphere/marathon/core/health/impl/HealthCheckActorTest.scala

```diff
@@ -143,6 +144,20 @@ class HealthCheckActorTest extends AkkaUnitTest {
       verifyNoMoreInteractions(f.tracker, f.driver, f.scheduler, f.killService)
     }
 
+    "task should always be killed if application doesn't set upgradeStrategy.minimumHealthCapacity" in {
+      val f = new Fixture
```

Comment:
I think we should add a test to specify behavior for app having a minimumHealthCapacity > 1 (explicitely defined)